diesel/src/sqlite/connection/raw.rs

```diff
@@ -17,8 +19,11 @@ impl RawConnection {
     pub fn establish(database_url: &str) -> ConnectionResult<Self> {
         let mut conn_pointer = ptr::null_mut();
         let database_url = try!(CString::new(database_url));
-        let connection_status =
-            unsafe { ffi::sqlite3_open(database_url.as_ptr(), &mut conn_pointer) };
+        let flags =
+            SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_NOMUTEX | SQLITE_OPEN_URI;
```

Comment:
Why are we specifying `OPEN_NOMUTEX` here? It is not a flag set by `sqlite3_open`, and it seems unrelated to your change.

Reply:
My understanding is that this flag is required to use the shared memory cache mode when opening a file with URI `file::memory:?cache=shared`.

Reply:
I'd prefer we hold off on making that change for the time being. We certainly can't just do it by default, but we can possibly expose an API for this in the future.